cmd/xl-storage-disk-id-check.go

```diff
@@ -568,34 +596,44 @@ func (p *xlStorageDiskIDCheck) StatInfoFile(ctx context.Context, volume, path st
 	}
 	defer done(&err)
 
-	return p.storage.StatInfoFile(ctx, volume, path, glob)
+	w := xioutil.NewDeadlineWorker(diskMaxTimeout)
+	rerr := w.Run(func() error {
+		stat, err = p.storage.StatInfoFile(ctx, volume, path, glob)
```

Comment:
Again, batch call - the glob can expand to thousands of entries. Feel free to add at a lower lever.